### ISSUE TITLE
Bug: no-restricted-globals does not look in global objects

### ISSUE THREAD

**Issue by barbeque-squared** [eslint/eslint#21121]
### Environment

Node version: v26.4.0
npm version: v11.18.0
Local ESLint version: v10.7.0 (Currently used)
Global ESLint version: Not found
Operating System: linux 7.1.3-arch1-1

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
import { defineConfig } from 'eslint/config'

export default defineConfig(
  {
    name: 'base: ban native fetch',
    files: ['**/*.js'],
    rules: {
      'no-restricted-globals': [
        'error',
        {
          globals: [
            {
              name: 'fetch',
              message: 'Use SomeCustomFetch instead',
            },
          ],
          checkGlobalObject: true,
          globalObjects: ['global'],
        },
      ],
      // temporary workaround
      // 'no-restricted-properties': [
      //   'error',
      //   {
      //     object: 'global',
      //     property: 'fetch',
      //     message: 'Use SomeCustomFetch instead',
      //   },
      // ],
    },
  }
)
```
</details>

```js
// this gets correctly detected
const x = await fetch('some/url')

// this should already get detected by checkGlobalObject: true
const z = await window.fetch('some/url')

// this should already get detected by checkGlobalObject: true + globalObjects: ['global']
const y = await global.fetch('some/url')
```

### What did you expect to happen?

Three `no-restricted-globals` errors should be reported, one each for lines 2, 5 and 8

### What actually happened?

Only the error at line 2 is reported

### Link to Minimal Reproducible Example

https://github.com/barbeque-squared/eslint-no-restricted-globals

### Participation

- [ ] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I did some `console.log`s after the `const globalScope = sourceCode.getScope(node);` in `lib/rules/no-restricted-globals.js`:
* I think `globalObjects` is correct since it logs `Set(4) { 'globalThis', 'self', 'window', 'global' }`
* But then `variable` is `null` for all cases except `globalThis`, and there it has `references: []` so that kills the next loop

I can get the rule to do what I want by duplicating the message to a `no-restricted-properties` rule on `global.fetch` (and `window.fetch`...), but I feel that this should be unnecessary.

**Comment by DMartens** [comment]
You are not specifying `window` or `global` as a [global variable](https://eslint.org/docs/latest/use/configure/language-options#specify-globals), as the rule [reports](https://eslint.org/play/#eyJ0ZXh0IjoiLypnbG9iYWwgZ2xvYmFsLCB3aW5kb3cgKi9cbi8qZXNsaW50IG5vLXJlc3RyaWN0ZWQtZ2xvYmFsczogW1wiZXJyb3JcIiwge1xuICAgIGdsb2JhbHM6IFtcImZldGNoXCJdLFxuICAgIGNoZWNrR2xvYmFsT2JqZWN0OiB0cnVlLFxuICAgIGdsb2JhbE9iamVjdHM6IFtcImdsb2JhbFwiXVxufV0qL1xuXG4vLyB0aGlzIGdldHMgY29ycmVjdGx5IGRldGVjdGVkXG5jb25zdCB4ID0gYXdhaXQgZmV0Y2goJ3NvbWUvdXJsJylcblxuLy8gdGhpcyBzaG91bGQgYWxyZWFkeSBnZXQgZGV0ZWN0ZWQgYnkgY2hlY2tHbG9iYWxPYmplY3Q6IHRydWVcbmNvbnN0IHogPSBhd2FpdCB3aW5kb3cuZmV0Y2goJ3NvbWUvdXJsJylcblxuLy8gdGhpcyBzaG91bGQgYWxyZWFkeSBnZXQgZGV0ZWN0ZWQgYnkgY2hlY2tHbG9iYWxPYmplY3Q6IHRydWUgKyBnbG9iYWxPYmplY3RzOiBbJ2dsb2JhbCddXG5jb25zdCB5ID0gYXdhaXQgZ2xvYmFsLmZldGNoKCdzb21lL3VybCcpXG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7fSwibGFuZ3VhZ2VPcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=) in all three cases if both `global` and `window` are declared as globals (in the example via a global directive on the first line).

**Comment by barbeque-squared** [comment]
> You are not specifying window or global as a [global variable](https://eslint.org/docs/latest/use/configure/language-options#specify-globals),

Thanks, this was the part I was missing. I did see the `/*global*/` comments on the [rule page](https://eslint.org/docs/latest/rules/no-restricted-globals), I just completely glossed over them because I wanted to use a config file anyway and didn't realize these were eslint-specific comments too.

Some other rules (eg `no-global-assign` or `no-native-reassign`) do appear to have a link to that configuration section, it might be useful to have that on this one too?

**Comment by DMartens** [comment]
Most users will get a lint error from the recommended `no-undef` rule that the variable is not defined and its documentation also links to specifying globals.
As such I do not think we need to add a link.
In the case of of `no-global-assign` (and its prior deprecated rule `no-native-reassign`) the link is to show how to specify writable globals which helps solving linting problems.
